Mage.Tests/src/test/java/org/mage/test/cards/abilities/other/GainAbilitiesTest.java

```diff
@@ -66,6 +66,63 @@ public void test_AttachmentUnique() {
                 ).count());
     }
 
+    @Test
+    public void testGainAbilityControlledSpells() {
+        removeAllCardsFromLibrary(playerA);
+        skipInitShuffling();
+
+        addCard(Zone.GRAVEYARD, playerA, "Hoarding Broodlord"); // gives Pestilent Spirit convoke
+        addCard(Zone.HAND, playerA, "Reanimate"); // to put Hoarding Broodlord in play
+        addCard(Zone.BATTLEFIELD, playerA, "Swamp", 1); // to cast Reanimate
+        addCard(Zone.BATTLEFIELD, playerA, "Firesong and Sunspeaker"); // gives Shock lifelink
+        addCard(Zone.LIBRARY, playerA, "Shock", 1); // to find with Hoarding Broodlord
+        addCard(Zone.HAND, playerA, "Covetous Urge"); // makes Pestilent Spirit castable from exile
+        addCard(Zone.BATTLEFIELD, playerA, "Island", 4); // to cast Covetous Urge
+        addCard(Zone.HAND, playerB, "Pestilent Spirit"); // gives Shock deathtouch
+        addCard(Zone.BATTLEFIELD, playerA, "Sol Ring"); // to cast Pestilent Spirit
+
+        castSpell(1, PhaseStep.PRECOMBAT_MAIN, playerA, "Reanimate", "Hoarding Broodlord"); // tap Swamp, lose 8 life, find Shock
+        waitStackResolved(1, PhaseStep.PRECOMBAT_MAIN);
+        castSpell(1, PhaseStep.PRECOMBAT_MAIN, playerA, "Covetous Urge", playerB); // tap four Islands, find Pestilent Spirit
+        waitStackResolved(1, PhaseStep.PRECOMBAT_MAIN);
+        castSpell(1, PhaseStep.PRECOMBAT_MAIN, playerA, "Pestilent Spirit"); // tap Sol Ring and Hoarding Broodlord
+        waitStackResolved(1, PhaseStep.PRECOMBAT_MAIN);
+        castSpell(1, PhaseStep.PRECOMBAT_MAIN, playerA, "Shock", "Firesong and Sunspeaker"); // convoke, lethal, gain 2 life
+
+        setStopAt(1, PhaseStep.BEGIN_COMBAT);
```

Comment:
miss `setStrictChooseMode(true);` -- must be added in all new tests

Reply:
If it's always required, is it possible to add it as default? Or would that break too many existing tests unless manually fixed?

Reply:
Yes, it must be in all tests someday, but there are still many tests with hidden AI choices (if you miss choice command then AI will choose). Also some old test with "can't play/pay" uses that AI hidden logic instead `checkPlayableAbility`.

Look at `assertWrongChoiceUsage`:
![shot_230624_082659](https://github.com/magefree/mage/assets/8344157/68c206bb-eeb8-4371-89bc-fc53fbbba909)